src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -529,6 +535,7 @@ public DetailedPackageMetadata PackageMetadata
                     OnPropertyChanged(nameof(PackageMetadata));
                     OnPropertyChanged(nameof(IsPackageDeprecated));
                     OnPropertyChanged(nameof(IsPackageVulnerable));
+                    OnPropertyChanged(nameof(IsPackageVulnerableOrDeprecated));
```

Comment:
It looks like we're doing the OnPropertyChanged call for this property, and some of the other ones, more than one time. The setter for PackageVulnerabilities calls OnPropertyChanged with the appropriate fields that depend on it. Since this setter for PackageMetadata sets the PackageVulnerabilities, and then again calls OnPropertyChanged for those fields, we're calling it twice. We should make sure the OnPropertyChanged events get fired only once for each of these properties. Separate which ones are dependent on PackageVulnerabilities and which ones are dependent on PackageMetadata, considering how the PackageMetadata setter also sets PackageVulnerabilities.

Reply:
Valid, I think I should be able to get away with only firing it in the PackageMetadata setter. It looks like a bunch of them are being set twice at the moment though.